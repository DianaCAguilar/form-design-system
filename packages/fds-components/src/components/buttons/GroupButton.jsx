import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

export const GroupButton = (props) => {
  const { className, isActive, Icon, label, ...rest } = props;

  const rootClass = cx('groupbtn', className, {
    active: isActive,
  });

  return (
    <button className={rootClass} {...rest}>
      {Icon && <Icon size="xs" className="groupbtn-icon" />}
      {label}
    </button>
  );
};

GroupButton.propTypes = {
  /** Displays active button style when true */
  isActive: PropTypes.bool,

  label: PropTypes.string,

  className: PropTypes.string,

  /** Specify a fds-icon component */
  Icon: PropTypes.func,
};

export default GroupButton;
