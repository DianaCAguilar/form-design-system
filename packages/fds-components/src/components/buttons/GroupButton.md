`<GroupButton />` should only be used inside `<ButtonGroup />`. This component controls it's own state. See above for example usage.