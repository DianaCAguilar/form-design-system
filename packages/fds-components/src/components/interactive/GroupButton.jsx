import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import baseElement from '../../util/baseElement';

export const GroupButton = (props) => {
  const { className, isActive, Icon, label, as, ...rest } = props;

  const Element = baseElement({ href: rest.href, as });

  const rootClass = cx('groupbtn', className, {
    'groupbtn--disabled': props.disabled,
    'groupbtn--active': isActive,
  });

  return (
    <Element className={rootClass} {...rest}>
      {Icon && <Icon size="xs" className="groupbtn-icon" />}
      {label}
    </Element>
  );
};

GroupButton.propTypes = {
  /** Displays active button style when true */
  isActive: PropTypes.bool,

  as: PropTypes.func,

  label: PropTypes.string,

  className: PropTypes.string,

  disabled: PropTypes.bool,

  /** Specify a fds-icon component */
  Icon: PropTypes.func,
};

export default GroupButton;
