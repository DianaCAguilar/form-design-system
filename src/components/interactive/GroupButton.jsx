import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';
import baseElement from '../../util/baseElement';

export const GroupButton = (props) => {
  const {
    isActive,
    Icon,
    label,
    Link,
    Wrapper,
    isFirstButton,
    isLastButton,
    ...rest
  } = props;

  const Element = baseElement({ href: rest.href, as: Link });

  const rootClass = cx('groupbtn', {
    'groupbtn--disabled': props.disabled,
    'groupbtn--active': isActive,
    'groupbtn--first': isFirstButton,
    'groupbtn--last': isLastButton,
  });

  return (
    <Wrapper>
      <Element {...rest} className={rootClass}>
        {Icon && <Icon size="xs" className="groupbtn-icon" />}
        {label}
      </Element>
    </Wrapper>
  );
};

GroupButton.defaultProps = {
  Wrapper: (props) => props.children,
};

GroupButton.propTypes = {
  /** Displays active button style when true */
  isActive: PropTypes.bool,

  /**
   * Pass **only** react-router `Link` here. You may **not**
   * pass anything else here: SFC, Class Component, etc (even
   * if they use react-router `Link` underneath the hood).
   */
  Link: PropTypes.func,

  label: PropTypes.string,

  disabled: PropTypes.bool,

  /** Specify a fds-icon component */
  Icon: PropTypes.func,

  /**
   * A component to wrap the entire group button. The
   * `wrapper` must render the children prop passed into it
   * to render the GroupButton.
   */
  Wrapper: PropTypes.elementType,

  /** For internal use: Assigns the first button styling. */
  isFirstButton: PropTypes.bool,

  /** For internal use: Assigns the last button styling. */
  isLastButton: PropTypes.bool,
};

export default GroupButton;
