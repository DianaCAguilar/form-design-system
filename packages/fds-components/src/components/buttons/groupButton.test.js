import React from 'react';
import { shallow } from 'enzyme';

import GroupButton from './GroupButton';

const renderComponent = (props) => shallow(<GroupButton {...props} />);

const Icon = () => (
  <span>ðŸŒ­</span>
)

describe('GroupButton component', () => {

  it('matches snapshot (set all props)', () => {
    const component = renderComponent({
      Icon,
      onClick: () => false,
      active: true,
      className: 'foo',
    });
    expect(component).toMatchSnapshot();
  });
});
