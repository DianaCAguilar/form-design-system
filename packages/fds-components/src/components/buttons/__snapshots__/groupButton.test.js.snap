// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupButton component matches snapshot (set all props) 1`] = `
<button
  className="groupbtn foo active"
  onClick={[Function]}
>
  <Icon
    className="margin--right--half"
    size="xs"
  />
</button>
`;
